cloud/azure/management/costs/mode/orphanresource.pm

```diff
@@ -0,0 +1,236 @@
+            api_version_override => "2022-01-01"
+            );
+        $self->{public_ips} = {orphan => 0, total => 0, name => ""};
+        foreach my $item (@{$items}) {
+            $self->{public_ips}->{total}++;;
+            next if (defined($self->{option_results}->{exclude_name}) && $self->{option_results}->{exclude_name} ne ''
+                     && $item->{name} =~ /$self->{option_results}->{exclude_name}/);
+            next if (defined($item->{properties}->{ipConfiguration}) && scalar($item->{properties}->{ipConfiguration}) != 0);
+            $self->{public_ips}->{orphan}++;
+            $self->{public_ips}->{name} .= $item->{name} . " ";
+        }
+        if ($self->{public_ips}->{orphan}) {
+            $output_error .= "Found " . $self->{public_ips}->{orphan} . " orphan public IP(s) ( " . $self->{public_ips}->{name} . ")\n";
+        }
+        else {
+            $output .= "No orphan public IP found on " . $self->{public_ips}->{total} . " total public IP(s)\n";
+        }
+    }
+
+    # orphan route tables
+    if (!defined($self->{option_results}->{no_route_tables})) {
+        $items = $options{custom}->azure_list_route_tables(
+            resource_group => $self->{option_results}->{resource_group},
+            api_version_override => "2022-01-01"
+            );
+        $self->{route_tables} = {orphan => 0, total => 0, name => ""};
+        foreach my $item (@{$items}) {
+            $self->{route_tables}->{total}++;;
+            next if (defined($self->{option_results}->{exclude_name}) && $self->{option_results}->{exclude_name} ne ''
+                     && $item->{name} =~ /$self->{option_results}->{exclude_name}/);
+            next if (defined($item->{properties}->{subnets}) && scalar($item->{properties}->{subnets}) != 0);
+            $self->{route_tables}->{orphan}++;
+            $self->{route_tables}->{name} .= $item->{name} . " ";
+        }
+        if ($self->{route_tables}->{orphan}) {
+            $output_error .= "Found " . $self->{route_tables}->{orphan} . " orphan route table(s) ( " . $self->{route_tables}->{name} . ")\n";
+        }
+        else {
+            $output .= "No orphan route table found on " . $self->{route_tables}->{total} . " total route table(s)\n";
+        }
+    }
+
+    # orphan snapshots
+    if (!defined($self->{option_results}->{no_snapshots})) {
+        $items = $options{custom}->azure_list_snapshots(
+            resource_group => $self->{option_results}->{resource_group},
+            api_version_override => "2021-12-01"
+            );
+        $self->{snapshots} = {orphan => 0, total => 0, name => ""};
+        foreach my $item (@{$items}) {
+            $self->{snapshots}->{total}++;;
+            next if (defined($self->{option_results}->{exclude_name}) && $self->{option_results}->{exclude_name} ne ''
+                     && $item->{name} =~ /$self->{option_results}->{exclude_name}/);
+            $self->{snapshots}->{orphan}++;
```

Comment:
Yes it's normal.
A snapshot is not attached to any other resource.
it's usefull to know if a forgotten snapshot is present. It can always been put un er exclusion (its name or whole snapshot type)

Reply:
Ok makes sense. Thank you!